src/plugins/intel_gpu/src/graph/rms.cpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "rms_inst.h"
+
+#include "primitive_type_base.h"
+#include "json_object.h"
+#include <string>
+
+namespace cldnn {
+GPU_DEFINE_PRIMITIVE_TYPE_ID(rms);
+
+layout rms_inst::calc_output_layout(rms_node const& node, kernel_impl_params const& impl_param) {
+    auto desc = impl_param.typed_desc<rms>();
+    auto input_layout = impl_param.get_input_layout();
+    auto output_type = desc->output_data_types[0].value_or(input_layout.data_type);
+    auto output_format = input_layout.format;
+
+    return layout(output_type, output_format, input_layout.get_tensor());
+}
+
+template<typename ShapeType>
+std::vector<layout> rms_inst::calc_output_layouts(rms_node const& /*node*/, kernel_impl_params const& impl_param) {
+    auto desc = impl_param.typed_desc<rms>();
+    auto input_layout = impl_param.get_input_layout();
+    auto output_type = desc->output_data_types[0].value_or(input_layout.data_type);
+    auto output_format = input_layout.format;
+    auto output_shape = input_layout.get<ShapeType>();
+
+    return { layout(output_shape, output_type, output_format) };
```

Comment:
why not `forward_input0_shape` ?

Reply:
I mistakenly thought that prim's `output_data_types` was not being handled in `forward_input0_shape`. I reverted changes again.